munet/mutest/userapi.py

```diff
@@ -742,20 +743,26 @@ def section(self, desc: str):
         self.oplogf("   section setting __in_section to True")
         self.__print_header(self.info.tag, desc, add_nl)
 
-    def step(self, target: str, cmd: str) -> str:
+    def step(
+        self,
+        target: str,
+        cmd: str,
+        output: bool = True,
+    ) -> str:
```

Comment:
In general creating these fire-and-forget processes is messy b/c there's no guarantee they will complete before the test exists (in which case they will probably be killed by the kernel, or have PIPE closed signals or something). For non-mutest uses one uses `popen` to start processes that you want to run in the background and you get back a `p` process object that you can kill or wait on later. I don't know if this is the right pattern to use for mutest though -- so maybe your way here is ok.

Reply:
I wonder about the actual use case. Might a new API that runs multiple commands simultaneously and waits for them all to complete would be a cleaner solution (`step_multi` or `step_parallel`)? It depends on the problem we're trying to solve I guess.

Reply:
The use case I have encountered twice now is the desire to create some sort of traffic generator on a node (or two nodes in the case of setting up a client/server pair situation) and then run a series of tests based on the results (perhaps on separate nodes). While it would probably be feasible to do this with some sort of new `waitstep_multi` or `waitstep_parallel`, such would require that API call to be responsible for multiple parallel steps plus multiple waiting steps which seems excessive. I figured that accepting the risk and running a few fire-and-forget cmd through a `step` (then using `match`/`wait` calls as usual but written by the tester to be robust to the state of the background process) was cleanest.

I do agree that letting the kernel kill the processes is a naive solution though, so perhaps the best solution would be to keep track of all fire-and-forget processes in a list and clean it up later when a node is being deleted?